Add network best practice document (#429)
harvester/docs: docs/networking/best-practice.md

  router(config)# ip route 192.168.200.0 255.255.255.0 ethernet0/2
  router(config)# end
  ```

  The route table would be like this:

  ```
  Router#show ip route
  Codes: C - connected, S - static, R - RIP, M - mobile, B - BGP
         D - EIGRP, EX - EIGRP external, O - OSPF, IA - OSPF inter area
         N1 - OSPF NSSA external type 1, N2 - OSPF NSSA external type 2
         E1 - OSPF external type 1, E2 - OSPF external type 2
         i - IS-IS, su - IS-IS summary, L1 - IS-IS level-1, L2 - IS-IS level-2
         ia - IS-IS inter area, * - candidate default, U - per-user static route
         o - ODR, P - periodic downloaded static route

  Gateway of last resort is not set

  C    192.168.200.0/24 is directly connected, Ethernet0/2.200
     10.0.0.0/24 is subnetted, 1 subnets
  C       10.10.0.0 is directly connected, Ethernet0/1
  C    192.168.0.0/24 is directly connected, Ethernet0/2
  C    192.168.100.0/24 is directly connected, Ethernet0/2.100
  ```

## Access Harvester Load Balancers from Different VLANs
The Harvester load balancer is divided into two types: VM load balancer and guest Kubernetes cluster load balancer. 

> 1. The load balancer IP of the VM load balancer is only exposed within the same network as the Harvester hosts, or in other words, the management network. To access the VM load balancer from outside the network, you have to prove the route from outside client to the management network. For example, if the VM load balancer `lb1` has obtained its load balancer IP var DHCP and you want to access it from the VM `VM5`, you need to add the following static route on the router to

Reviewer: ```suggestion
1. The load balancer IP of the VM load balancer is only exposed within the same network as the Harvester hosts, or in other words, the management network. To access the VM load balancer from outside the management network, you need to add a static route on the router. For example, if the VM load balancer `lb1` has obtained its load balancer IP var DHCP and you want to access it from the VM `VM5`, you need to add the following static routes:
```

Reviewer: Not sure if `IP var DHCP` is the correct term

Author: Sorry, it should be `IP via DHCP`.

  ```
  router(config)# ip route 10.10.0.0 255.255.255.0 ethernet0/1
  router(config)# ip route 192.168.0.0 255.255.255.0 ethernet0/2
  ```

> 2. The load balancer IP of the guest Kubernetes cluster load balancer is exposed within the VM network. In the diagram above, the guest cluster `demo` is within the VM network `vlan200` because the VMs consist of the guest cluster are in the `vlan200`. Thus, the guest Kubernetes cluster load balancer `lb2` is exposed within the VM network `vlan200`. There are three scenarios to explain how to access `lb2` if it has obtained the load balancer IP via DHCP:

Reviewer: ```suggestion
2. The load balancer IP of the guest Kubernetes cluster load balancer is exposed within the VM network. In the diagram above, the guest cluster `demo` is within the VM network `vlan200` because the VMs consisting of the guest cluster are in the `vlan200`. Thus, the guest Kubernetes cluster load balancer `lb2` is exposed within the VM network `vlan200`. There are three scenarios to explain how to access `lb2` if it has obtained the load balancer IP via DHCP:
```

   - You can access it from the VM `VM3` and `VM4` directly because they are in the `vlan200`. 
   - You can also access it from the VMs in other VM network directly because of the `A router on a stick` configuration.

>    - You can access it from the Harvester hosts, or in other words, the management network by adding the static route on the router. 

Reviewer: ```suggestion
   - You can access it from the Harvester hosts, or in other words, the management network by adding the following static routes on the router. 
```

     ```
     router(config)# ip route 10.10.0.0 255.255.255.0 ethernet0/1
     router(config)# ip route 192.168.200.0 255.255.255.0 ethernet0/2
     ```

:::note

> Except static route used above, you can also use dynamic routing protocols such as RIP, BGP, OSPF and ISIS according to your network planning and requirements.

Reviewer: ```suggestion
Except for the static routes above, you can also use dynamic routing protocols such as RIP, BGP, OSPF, and ISIS according to your network planning and requirements.
```

  router(config)# ip route 192.168.200.0 255.255.255.0 ethernet0/2
  router(config)# end
  ```

  The route table would be like this:

  ```
  Router#show ip route
  Codes: C - connected, S - static, R - RIP, M - mobile, B - BGP
         D - EIGRP, EX - EIGRP external, O - OSPF, IA - OSPF inter area
         N1 - OSPF NSSA external type 1, N2 - OSPF NSSA external type 2
         E1 - OSPF external type 1, E2 - OSPF external type 2
         i - IS-IS, su - IS-IS summary, L1 - IS-IS level-1, L2 - IS-IS level-2
         ia - IS-IS inter area, * - candidate default, U - per-user static route
         o - ODR, P - periodic downloaded static route

  Gateway of last resort is not set

  C    192.168.200.0/24 is directly connected, Ethernet0/2.200
     10.0.0.0/24 is subnetted, 1 subnets
  C       10.10.0.0 is directly connected, Ethernet0/1
  C    192.168.0.0/24 is directly connected, Ethernet0/2
  C    192.168.100.0/24 is directly connected, Ethernet0/2.100
  ```

## Access Harvester Load Balancers from Different VLANs
The Harvester load balancer is divided into two types: VM load balancer and guest Kubernetes cluster load balancer. 

> 1. The load balancer IP of the VM load balancer is only exposed within the same network as the Harvester hosts, or in other words, the management network. To access the VM load balancer from outside the network, you have to prove the route from outside client to the management network. For example, if the VM load balancer `lb1` has obtained its load balancer IP via DHCP and you want to access it from the VM `VM5`, you can add the following static route.

Reviewer: ```suggestion
1. The load balancer IP of the VM load balancer is only exposed within the same network as the Harvester hosts, or in other words, the management network. To access the VM load balancer from outside the management network, you need to add a static route on the router. For example, if the VM load balancer `lb1` has obtained its load balancer IP via DHCP and you want to access it from the VM `VM5`, you need to add the following static routes:
```

Reviewer: This comment hasn't been applied. The rest are good to me.

Author: The expression seems not to be accurate.

  ```
  router(config)# ip route 10.10.0.0 255.255.255.0 ethernet0/1
  router(config)# ip route 192.168.0.0 255.255.255.0 ethernet0/2
  ```

> 2. The load balancer IP of the guest Kubernetes cluster load balancer is exposed within the VM network. In the diagram above, the guest cluster `demo` is within the VM network `vlan200` because the VMs consisting of the guest cluster are in the `vlan200`. Thus, the guest Kubernetes cluster load balancer `lb2` is exposed within the VM network `vlan200`. There are three scenarios to explain how to access `lb2` if it has obtained the load balancer IP via DHCP:
>    - You can access it from the VM `VM3` and `VM4` directly because they are in the `vlan200`. 
>    - You can also access it from the VMs in other VM network directly because of the `A router on a stick` configuration.
>    - You can access it from the Harvester hosts, or in other words, the management network by adding the following static routes on the router. 

Reviewer 2: ```suggestion
1. The load balancer IP of the guest Kubernetes cluster load balancer is exposed within the VM network. In the diagram above, the guest cluster `demo` is within the VM network `vlan200` because the VMs consisting of the guest cluster are in the `vlan200`. Thus, the guest Kubernetes cluster load balancer `lb2` is exposed within the VM network `vlan200`. There are three scenarios to explain how to access `lb2` if it has obtained the load balancer IP via DHCP:
   - You can access it from the VM `VM3` and `VM4` directly because they are in the `vlan200`. 
   - You can also access it directly from the VMs in other VM networks because of the `A router on a stick` configuration.
   - You can access it from the Harvester hosts, or in other words, the management network, by adding the following static routes on the router.  
```

     ```
     router(config)# ip route 10.10.0.0 255.255.255.0 ethernet0/1
     router(config)# ip route 192.168.200.0 255.255.255.0 ethernet0/2
     ```

:::note

> Except for the static routes above, you can also use dynamic routing protocols such as RIP, BGP, OSPF, and ISIS according to your network planning and requirements.

Reviewer 2: ```suggestion
Except for the static routes above, you can use dynamic routing protocols such as RIP, BGP, OSPF, and ISIS according to your network planning and requirements.
```
